backend/src/search/views.py

```diff
@@ -34,6 +38,28 @@ async def create_search_task(file: UploadFile, user_id: int) -> dict:
     return {"message": "successful upload", "filename": file.filename}
 
 
+@router.get("/search/{search_id}/status")
+async def get_search_status(search_id: int, db_session: Session):
+    try:
+        search = SearchRepository.find_by_id(db_session, search_id)
+        task_status = AsyncResult(search.task_id).status
+        if (
+            search.status == SearchTaskStatus.PENDING
+            and task_status not in states.UNREADY_STATES
+        ):
+            if task_status in states.EXCEPTION_STATES:
+                search = SearchRepository.update(
+                    db_session, search, {"status": SearchTaskStatus.ERROR}
+                )
+            elif task_status == states.SUCCESS:
+                search = SearchRepository.update(
+                    db_session, search, {"status": SearchTaskStatus.READY}
+                )
+    except Exception:
+        raise HTTPException(500, detail="Internal server error.")
+    return StatusResponseModel(status=search.status)
+
+
```

Comment:
its okay, but it could use guard statements if u are willing to re-factorize, its hard to read

Reply:
Please, check if it's better now